Add doc for SCIM integration using entity types. (#1366)
FusionAuth/fusionauth-site: site/docs/v1/tech/core-concepts/entity-management.adoc

* An example link:/docs/v1/tech/oauth/#example-client-credentials-grant[client credentials grant using Entities].
* The link:/docs/v1/tech/apis/entity-management/[Entity Management APIs].

== SCIM Integration
[NOTE.since]
====
This functionality has been available since 1.36.0
====

// TODO: need screenshots

> When configuring FusionAuth to accept SCIM requests, you must create a SCIM Server Entity and a SCIM Client Entity. The Entity Types can be created by selecting the down arrow next to the add icon icon:plus[role=ui-button green,type=fas] on the link:#entity-types[Entity Type homepage]. However, by default FusionAuth ships with the two SCIM Entity Types provided. An external SCIM Client must be of the Entity Type SCIM Client and have the associated permissions enabled to successfully call authenticated link:/docs/v1/tech/apis/scim[SCIM API endpoints].

Reviewer: ```suggestion
When configuring FusionAuth to accept SCIM requests, you must create a SCIM Server Entity and a SCIM Client Entity.
These entities will complete the Client Credentials grant which will provide the access token which is used to authenticate calls to the SCIM endpoints.
These entities must be of the Entity Type configured in the Tenant SCIM configuration. 
They also must have the SCIM permissions granted to successfully call link:/docs/v1/tech/apis/scim[SCIM API endpoints] requiring authentication.

The necessary Entity Types can be created by selecting the down arrow next to the add icon icon:plus[role=ui-button green,type=fas] on the Entity Type homepage.
However, by FusionAuth ships with the two SCIM Entity Types that you can use.
```

> When configuring FusionAuth to accept SCIM requests, you must create a SCIM Server Entity and a SCIM Client Entity. The Entity Types can be created by selecting the down arrow next to the add icon icon:plus[role=ui-button green,type=fas] on the link:#entity-types[Entity Type homepage]. However, by default FusionAuth ships with the two SCIM Entity Types provided. An external SCIM Client must be of the Entity Type SCIM Client and have the associated permissions enabled to successfully call authenticated link:/docs/v1/tech/apis/scim[SCIM API endpoints].

Author: Yeah, if you look at demo.fusionauth.io they are there.

> When configuring FusionAuth to accept SCIM requests, you must create a SCIM Server Entity and a SCIM Client Entity. The Entity Types can be created by selecting the down arrow next to the add icon icon:plus[role=ui-button green,type=fas] on the link:#entity-types[Entity Type homepage]. However, by default FusionAuth ships with the two SCIM Entity Types provided. An external SCIM Client must be of the Entity Type SCIM Client and have the associated permissions enabled to successfully call authenticated link:/docs/v1/tech/apis/scim[SCIM API endpoints].

Author: If you are doing a kickstart install then I don't think they are created.

> When configuring FusionAuth to accept SCIM requests, you must create a SCIM Server Entity and a SCIM Client Entity. The Entity Types can be created by selecting the down arrow next to the add icon icon:plus[role=ui-button green,type=fas] on the link:#entity-types[Entity Type homepage]. However, by default FusionAuth ships with the two SCIM Entity Types provided. An external SCIM Client must be of the Entity Type SCIM Client and have the associated permissions enabled to successfully call authenticated link:/docs/v1/tech/apis/scim[SCIM API endpoints].

Reviewer: There they are!

Makes sense they wouldn't be installed with Kickstart, thanks.

==== [FusionAuth Default] SCIM Client
The default SCIM Client Entity Type is the external SCIM client calling link:/docs/v1/tech/apis/scim[SCIM API endpoints].

==== [FusionAuth Default] SCIM Server
The default SCIM Server Entity Type is FusionAuth acting as the SCIM server and defines the available permissions for calling link:/docs/v1/tech/apis/scim[SCIM API endpoints].

> ==== [FusionAuth Default] SCIM Server Permissions

Reviewer: ```suggestion
==== SCIM Server Permissions
```

These aren't default, correct? If I wanted to create a new SCIM Entity Type, I'd have to use these same permissions.

Author: Well, they are created as part of the default scim server entity type. Wasn't sure of the best way to document that.

Author: Yes, that is correct. I see your point. I'll remove the default text.

> Each permission enables or disables access to an API endpoint.

Reviewer: ```suggestion
Each permission enables or disables access to an API endpoint.
These are required for any Entity Type you create which will be used to create Entities for SCIM authentication.

When you are creating an Entity representing a SCIM client, you must select one or more of these permissions, which will control which SCIM APIs can be called. 
```

> * scim:enterprise:user:create - link:/docs/v1/tech/apis/scim/SCIMEnterpriseUser#create-an-enterpriseuser[Create an EnterpriseUser]

Author: fixed
